tests/components/light/test_init.py

```diff
@@ -18,479 +18,479 @@
+    assert SERVICE_TURN_OFF == call.service
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        # Test turn_off
-        turn_off_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TURN_OFF)
+    # Test toggle
+    toggle_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TOGGLE)
 
-        common.turn_off(
-            self.hass, entity_id="entity_id_val", transition="transition_val"
-        )
+    await common.async_toggle(
+        hass, entity_id="entity_id_val", transition="transition_val"
+    )
 
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
 
-        assert 1 == len(turn_off_calls)
-        call = turn_off_calls[-1]
+    assert len(toggle_calls) == 1
+    call = toggle_calls[-1]
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TURN_OFF == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TOGGLE
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        # Test toggle
-        toggle_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TOGGLE)
 
-        common.toggle(self.hass, entity_id="entity_id_val", transition="transition_val")
+async def test_services(hass):
+    """Test the provided services."""
+    platform = getattr(hass.components, "test.light")
 
-        self.hass.block_till_done()
+    platform.init()
+    assert await async_setup_component(
+        hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
+    )
+    await hass.async_block_till_done()
 
-        assert 1 == len(toggle_calls)
-        call = toggle_calls[-1]
+    ent1, ent2, ent3 = platform.ENTITIES
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TOGGLE == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    # Test init
+    assert light.is_on(hass, ent1.entity_id)
+    assert not light.is_on(hass, ent2.entity_id)
+    assert not light.is_on(hass, ent3.entity_id)
 
-    def test_services(self):
-        """Test the provided services."""
-        platform = getattr(self.hass.components, "test.light")
+    # Test basic turn_on, turn_off, toggle services
+    await common.async_turn_off(hass, entity_id=ent1.entity_id)
+    await common.async_turn_on(hass, entity_id=ent2.entity_id)
 
-        platform.init()
-        assert setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
 
-        ent1, ent2, ent3 = platform.ENTITIES
+    assert not light.is_on(hass, ent1.entity_id)
+    assert light.is_on(hass, ent2.entity_id)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)